flake.nix

```diff
@@ -1,63 +1,60 @@
 {
   inputs = {
-    # Candidate channels
-    #   - https://discourse.nixos.org/t/differences-between-nix-channels/13998
-    # How to update the revision
-    #   - `nix flake update --commit-lock-file` # https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake-update.html
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
-    flake-utils.url = "github:numtide/flake-utils";
+    # nodejs-18_x is deprecated since nixos-25.05
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
   };
 
   outputs =
     {
       self,
       nixpkgs,
-      flake-utils,
     }:
-    flake-utils.lib.eachDefaultSystem (
-      system:
-      let
-        pkgs = nixpkgs.legacyPackages.${system};
-      in
-      {
-        formatter = pkgs.nixfmt-rfc-style;
-        devShells.default =
-          with pkgs;
-          mkShell {
-            buildInputs = [
-              # https://github.com/NixOS/nix/issues/730#issuecomment-162323824
+    let
+      inherit (nixpkgs) lib;
+      forAllSystems = lib.genAttrs lib.systems.flakeExposed;
+    in
+    {
+      # formatter = forAllSystems (system: nixpkgs.legacyPackages.${system}.nixfmt-tree); # nixfmt-tree is available since nixos-25.05
+      devShells = forAllSystems (
+        system:
+        let
+          pkgs = nixpkgs.legacyPackages.${system};
+        in
+        {
+          default = pkgs.mkShellNoCC {
+            buildInputs = (
+              with pkgs;
+              [
+                # https://github.com/NixOS/nix/issues/730#issuecomment-162323824
+                bashInteractive
+                # Maintain this repo
+                nil
+                nixfmt-rfc-style
+                dprint
+                typos
+                go-task
+                shellcheck
+                shfmt
+                hostname
+              ]
+            );
+          };
+
+          lila = pkgs.mkShell {
+            buildInputs = with pkgs; [
               bashInteractive
 
-              # Maintain this repo
-              nil
-              nixfmt-rfc-style
-              dprint
-              typos
-              go-task
-              shellcheck
-              shfmt
-              hostname
+              sbt
+              nodejs-18_x
+              nodejs-18_x.pkgs.yarn
```

Comment:
https://github.com/WandererXII/lishogi/blob/38c5201cc5ffb9a056444a17092fae89a5d0e1d3/package.json#L17

Upstream is now using pnpm